### PR TITLE
fix(errors): Better error messages

### PR DESCRIPTION
**Pull request by marchello2000**
* Improve error messages in FindImage stage when images are inferred from subsequent deploys
* Don't surface intermediate errors during SpEL evaluation

### CODE REVIEW

**Comment by ajordens** [comment]
LGTM

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy [hunk 1]
```diff
@@ -205,14 +207,19 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
...
-            throw new IllegalStateException("Missing images in $stillUnresolved.value")
-          }
-        } else {
-          throw new IllegalStateException("Missing images in $unresolved.value")
+          unresolved = imageSummaries.findResults { it.value == null ? it.key : null }
         }
       }
+
+      if (unresolved) {
+        def errorMessage = "Missing image '${searchNames[0]}' in regions: $unresolved.value"
```
Comment (robzienert) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy]: We don't need to wrap `${unresolved.value}`?

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.java [hunk 2]
```diff
@@ -62,11 +62,16 @@ public ContextParameterProcessor(ContextFunctionConfiguration contextFunctionCon
   }
 
   public Map<String, Object> process(Map<String, Object> source, Map<String, Object> context, boolean allowUnknownKeys) {
+    ExpressionEvaluationSummary summary = new ExpressionEvaluationSummary();
+
+    return process(source, context, allowUnknownKeys, summary);
+  }
+
+  public Map<String, Object> process(Map<String, Object> source, Map<String, Object> context, boolean allowUnknownKeys, ExpressionEvaluationSummary summary) {
```
Comment (robzienert) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.java]: Wouldn't mind seeing the arguments get wrapped so we're not going beyond 120.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt [hunk 3]
```diff
@@ -60,12 +62,28 @@ interface ExpressionAware {
...
+    // context. Otherwise, it's very confusing in the UI because the value is clearly correctly evaluated but
+    // the error is still shown
+    if (hasFailedExpressions()) {
+      val failedExpressions = this.context[PipelineExpressionEvaluator.SUMMARY] as MutableMap<String, *>
+
+      failedExpressions.keys.forEach( { expressionKey ->
+        if (evalSummary.wasAttempted(expressionKey) && !evalSummary.hasFailed(expressionKey)) {
+          failedExpressions.remove(expressionKey)
+        }
+      })
```
Comment (robzienert) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt]: Just a small little thing here, final arguments in Kotlin into a method don't need to be wrapped in parens.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy (hunk 1 above)
Comment (marchello2000) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy]: good catch, thanks!

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.java (hunk 2 above)
Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.java]: 👍

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt (hunk 3 above)
Comment (marchello2000) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt]: thanks!

**Pull request merged**